src/wallet/wallet.h

```diff
@@ -647,12 +648,66 @@ class LelantusJoinSplitBuilder;
 //static boost::signals2::signal<void (CWallet *wallet)> UnlockWallet;
 extern boost::signals2::signal<void (CWallet *wallet)> UnlockWallet;
 
+class CTransparentTxout;
+enum AbstractTxoutType {
+    Transparent
+};
+
+class CTransparentTxout {
+public:
+    static bool IsTransparentTxout(const CTxOut& txout);
+
+    CTransparentTxout() = default;
+    CTransparentTxout(COutPoint outpoint, CTxOut txout): outpoint(outpoint), txout(txout), _isMockup(true) {}
```

Comment:
Although we might not apply this comment right now. But it worth keep this in mind. 

I personally don't like to have test variables in production code. IMHO we can use something like test macro to eliminate test code from production code. 

For example this came up in my search: 

https://github.com/apache/incubator-resilientdb/blob/0541fe43bd1bbd69ecac5a1bc0d0cdd5ac04bc56/platform/statistic/stats.cpp#L44-L46

```
#ifdef TEST_MODE
  monitor_sleep_time_ = 1;
#endif

```